test/integration/analytical_rosenbrock.cpp

```diff
@@ -12,120 +12,156 @@
 
 using SparseMatrixTest = micm::SparseMatrix<double>;
 
-auto rosenbrock = micm::CpuSolverBuilder<micm::RosenbrockSolverParameters>(
+auto rosenbrock_2stage = micm::CpuSolverBuilder<micm::RosenbrockSolverParameters>(
+    micm::RosenbrockSolverParameters::TwoStageRosenbrockParameters());
+auto rosenbrock_3stage = micm::CpuSolverBuilder<micm::RosenbrockSolverParameters>(
     micm::RosenbrockSolverParameters::ThreeStageRosenbrockParameters());
+auto rosenbrock_4stage = micm::CpuSolverBuilder<micm::RosenbrockSolverParameters>(
+    micm::RosenbrockSolverParameters::FourStageRosenbrockParameters());
+auto rosenbrock_4stage_da = micm::CpuSolverBuilder<micm::RosenbrockSolverParameters>(
+    micm::RosenbrockSolverParameters::FourStageDifferentialAlgebraicRosenbrockParameters());
+auto rosenbrock_6stage_da = micm::CpuSolverBuilder<micm::RosenbrockSolverParameters>(
+    micm::RosenbrockSolverParameters::SixStageDifferentialAlgebraicRosenbrockParameters());
 auto backward_euler = micm::CpuSolverBuilder<micm::BackwardEulerSolverParameters>(micm::BackwardEulerSolverParameters());
 
 TEST(AnalyticalExamples, Troe)
 {
-  test_analytical_troe(
-      rosenbrock, [](auto& state) {}, [](auto& state) {});
-  test_analytical_troe(
-      backward_euler, [](auto& state) {}, [](auto& state) {}, 1e-4);
+  test_analytical_troe(rosenbrock_2stage, 1e-7);
+  test_analytical_troe(rosenbrock_3stage);
+  test_analytical_troe(rosenbrock_4stage);
+  test_analytical_troe(rosenbrock_4stage_da);
+  test_analytical_troe(rosenbrock_6stage_da);
+  test_analytical_troe(backward_euler, 1e-4);
```

Comment:
Should all the backward euler solver test be put to a separate file instead?

Reply:
Yeah, probably. I'll do that